### PR TITLE
Add registration fallback on login

### PR DESCRIPTION
**Pull request by antho1404**
If a user try to login and no account is found, fallback on the registration.

Screen Shot 2020-04-16 at 17 35 26

### CODE REVIEW

File: packages/cli/src/commands/login.ts [hunk 1]
```diff
@@ -25,7 +25,18 @@ export default class Login extends Command {
...
     }
   }
+
+  async register(email: string) {
+    this.log('')
+    this.warn(`${email} is not registered yet.`)
+    const { signup } = await prompt([{ name: 'signup', message: 'Would you like to create an account?', type: 'confirm' }])
+    if (!signup) return
+    const { password } = await prompt([{ name: 'password', message: `Password for your account ${email} ?`, type: 'password' }])
+    const { user } = await register(this.config.configDir, email, password)
```
Comment (NicolasMahe) [inline packages/cli/src/commands/login.ts:39]: if this function throws an error it is never displayed to the user and the command stop without more info.
In my case, the password was not accepted by firebase (I guess too short):
```
➜  liteflow-js git:(feature/register-user-login) ./packages/cli/bin/run login
? Your email dwdw@dwd.dw
? Your password [hidden]

 ›   Warning: dwdw@dwd.dw is not registered yet.
? Would you like to create an account? Yes
? Password for your account dwdw@dwd.dw ? [hidden]
```

Comment (antho1404) [reply packages/cli/src/commands/login.ts]: Fixed

**Comment by antho1404** [comment]
I added the flag `--register` in order to only register

File: packages/cli/src/commands/login.ts [hunk 2]
```diff
@@ -17,13 +18,31 @@ export default class Login extends Command {
...
+        const { signup } = await prompt([{ name: 'signup', message: 'Would you like to create an account?', type: 'confirm' }])
+        if (!signup) return
+        return this.register(email)
+      }
+      this.error(e.message)
+    }
+  }
+
+  async register(email: string) {
+    const { password } = await prompt([{ name: 'password', message: `Password for your account ${email} ?`, type: 'password' }])
```
Comment (NicolasMahe) [inline packages/cli/src/commands/login.ts:42]:
```suggestion
    const { password } = await prompt([{ name: 'password', message: `Your password`, type: 'password' }])
```

File: packages/cli/src/commands/login.ts [hunk 3]
```diff
@@ -17,13 +18,31 @@ export default class Login extends Command {
...
+    try {
+      const { user } = await register(this.config.configDir, email, password)
+      this.log(`Successfully logged in with the email: ${user.email}`)
```
Comment (NicolasMahe) [inline packages/cli/src/commands/login.ts:45]:
```suggestion
      this.log(`Successfully registered and logged in with the email: ${user.email}`)
```

**Pull request merged**